common/src/main/java/com/wynntils/core/notifications/NotificationManager.java

```diff
@@ -29,6 +43,26 @@ public static MessageContainer queueMessage(TextRenderTask message) {
 
         WynntilsMod.info("Message Queued: " + message);
         MessageContainer msgContainer = new MessageContainer(message);
+        String messageText = message.getText();
+
+        for (Pair<String, MessageContainer> cachedMessagePair : cachedMessageSet) {
+            String checkableMessage = cachedMessagePair.a();
+            if (messageText.equals(checkableMessage)) {
+                MessageContainer matchedContainer = cachedMessagePair.b();
+                matchedContainer.incrementMessageCount();
+
+                // Overlay is not enabled, send in chat
+                if (!GameNotificationOverlayFeature.INSTANCE.isEnabled()) {
```

Comment:
You moved this here. But!. What if we don't have cached messages? Shouldn't we handle this case after the for loop as well?

Reply:
it is handled after the for loop? see line 69-71.